firefox-ios/Client/Frontend/Autofill/Address/AddressListViewModel.swift

```diff
@@ -192,6 +197,26 @@ class AddressListViewModel: ObservableObject, FeatureFlaggable {
         ))
     }
 
+    func removeConfimationButtonTap() {
+        if case .edit(let address) = destination {
+            addressProvider.deleteAddress(id: address.id) { [weak self] result in
+                guard let self else { return }
+                DispatchQueue.main.async {
+                    switch result {
+                    case .success:
+                        self.presentToast?(.removed)
+                    case .failure:
+                        // TODO: FXIOS-9269 Create and add error toast for address saving failure
+                        break
+                    }
+                    self.toggleEditMode()
+                    self.destination = nil
+                    self.fetchAddresses()
```

Comment:
Note: weak self or unowned based on the preference?

Reply:
Yes. There's guarding against weak self at like 203.